staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/conversion/converter.go

```diff
@@ -76,17 +80,27 @@ func (m *CRConverterFactory) NewConverter(crd *apiextensionsv1.CustomResourceDef
 
 	// Determine whether we should expect to be asked to "convert" autoscaling/v1 Scale types
 	convertScale := false
+	selectableFields := map[schema.GroupVersion]sets.Set[string]{}
 	for _, version := range crd.Spec.Versions {
+		gv := schema.GroupVersion{Group: crd.Spec.Group, Version: version.Name}
 		if version.Subresources != nil && version.Subresources.Scale != nil {
 			convertScale = true
 		}
+		fieldPaths := sets.New[string]()
+		if utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceFieldSelectors) {
+			for _, sf := range version.SelectableFields {
+				fieldPaths.Insert(strings.TrimPrefix(sf.JSONPath, "."))
+			}
+		}
+		selectableFields[gv] = fieldPaths
```

Comment:
expected this inside the featuregate block.

Reply:
Fixed